src/components/Contacts/ContactFilters/ContactFilters.tsx

```diff
@@ -1,35 +1,211 @@
-import { colors } from '@material-ui/core';
-import React from 'react';
-import { useContactFiltersLazyQuery } from '../../../../pages/accountLists/[accountListId]/ContactFilters.generated';
+import {
+  Box,
+  BoxProps,
+  Button,
+  CircularProgress,
+  Collapse,
+  IconButton,
+  List,
+  ListItem,
+  ListItemText,
+  Slide,
+  styled,
+  Typography,
+  useTheme,
+} from '@material-ui/core';
+import { ArrowBackIos, ArrowForwardIos } from '@material-ui/icons';
+import { isArray } from 'lodash';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ContactFilterGroup } from '../../../../graphql/types.generated';
+import { Filter } from '../../Shared/Filters/Filter';
+import { FilterListItem } from '../../Shared/Filters/FilterListItem';
+import { FilterListItemShowAll } from '../../Shared/Filters/FilterListItemShowAll';
+import { useContactFiltersQuery } from './ContactFilters.generated';
+
+const FilterHeader = styled(Box)(({ theme }) => ({
+  padding: theme.spacing(2),
+  borderBottom: '1px solid',
+  borderBottomColor: theme.palette.grey[200],
+}));
+
+const FilterList = styled(List)(({ theme }) => ({
+  '& .MuiListItemIcon-root': {
+    minWidth: '37px',
+  },
+  '& .FilterListItemMultiselect-root': {
+    marginBottom: theme.spacing(4),
+  },
+}));
+
+const LinkButton = styled(Button)(() => ({
+  minWidth: 0,
+  textTransform: 'none',
+}));
 
 interface Props {
   accountListId: string;
 }
 
-export const ContactFilters: React.FC<Props> = ({ accountListId }: Props) => {
-  const [
-    loadContactFilters,
-    { data, loading, error },
-  ] = useContactFiltersLazyQuery({
+export const ContactFilters: React.FC<Props & BoxProps> = ({
+  accountListId,
+  ...boxProps
+}) => {
+  const theme = useTheme();
+  const { t } = useTranslation();
+
+  const { data, loading, error } = useContactFiltersQuery({
     variables: { accountListId },
   });
 
+  const [selectedGroup, showGroup] = useState<null | ContactFilterGroup>(null);
+  const [selectedFilters, setSelectedFilters] = useState({});
+  const [showAll, setShowAll] = useState(false);
+
+  const updateSelectedFilter = (name: string, value) => {
+    if (value && !(isArray(value) && value.length == 0))
+      setSelectedFilters((prev) => {
+        return { ...prev, [name]: value };
+      });
+    else
+      setSelectedFilters((prev) => {
+        const updated = { ...prev };
+        delete updated[name];
+        return updated;
+      });
+  };
+
+  const getSelectedFilters = (group: ContactFilterGroup) =>
+    group.filters.filter((value) => selectedFilters[value.name]);
+  const isGroupVisible = (group: ContactFilterGroup) =>
```

Comment:
Using updated type from `Filter.ts`:
```suggestion
  const isGroupVisible = (group: FilterGroup) =>
```